presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/retry/RetryDriver.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.benchmark.retry;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.benchmark.framework.QueryException;
+import io.airlift.units.Duration;
+
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Predicate;
+
+import static com.google.common.base.Throwables.throwIfUnchecked;
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+import static java.lang.Math.pow;
+import static java.lang.Thread.currentThread;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class RetryDriver<E extends RuntimeException>
+{
+    private static final Logger log = Logger.get(RetryDriver.class);
+
+    private final int maxAttempts;
+    private final Duration minBackoffDelay;
+    private final Duration maxBackoffDelay;
+    private final double scaleFactor;
+    private final Predicate<E> retryPredicate;
+    private final Class<E> exceptionClass;
```

Comment:
As discussed, this is no needed. Exception Class filter can be achieved by retryPredicate